### PR TITLE
Enable purge protection for app configuration stores

### PR DESCRIPTION
**Pull request by henrikwh**
# Overview/Summary
Queries testing that
1) purge protection has been enabled for app configuration store.
2) free version should not be used

## As part of this pull request I have

- [x] Read the [Contribution Guide](https://azure.github.io/Azure-Proactive-Resiliency-Library-v2/contributing) and ensured this PR is compliant with the guide
- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/pulls)
- [x] Associated it with relevant [GitHub Issues](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/issues) or ADO Work Items (Internal Only)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/tree/main)
- [x] Ensured PR tests are passing
- [x] Performed testing and provided evidence (e.g. screenshot of output) for any changes associated to ARG queries

Test for purge protection
image

Test for free version
image

- [x] Updated relevant and associated documentation (e.g. Contribution Guide, Docs etc.)

### CODE REVIEW

File: azure-resources/AppConfiguration/configurationStores/recommendations.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    With Purge protection enabled, soft deleted stores can't be purged in the retention period. If disabled, the soft deleted store can be purged before the retention period expires.
+  potentialBenefits: Prevent accidental deletion of configuration stores.
+  pgVerified: false
+  publishedToLearn: false
+  publishedToAdvisor: false
+  automationAvailable: arg
+  tags: null
+  learnMoreLink:
+    - name: Purge protection
+      url: "https://learn.microsoft.com/en-us/azure/azure-app-configuration/concept-soft-delete#purge-protection"
```
Comment (ejhenry) [inline azure-resources/AppConfiguration/configurationStores/recommendations.yaml:18]:
```suggestion
      url: "https://learn.microsoft.com/en-us/azure/azure-app-configuration/concept-soft-delete#purge-protection"

```

File: azure-resources/AppConfiguration/configurationStores/kql/2102a57a-a056-4d5e-afe5-9df9f92177ca.kql [hunk 2]
```diff
@@ -0,0 +1,6 @@
+// Azure Resource Graph Query
+// Purge protection should be enabled for App Configuration stores to prevent accidental deletion of configuration data.
```
Comment (ejhenry) [inline azure-resources/AppConfiguration/configurationStores/kql/2102a57a-a056-4d5e-afe5-9df9f92177ca.kql:2]:
```suggestion
// Upgrade to App Configuration Standard tier
```

**Review by ejhenry — APPROVED** [review]
LGTM 👍

**Review by oZakari — APPROVED** [review]
LGTM, thanks @henrikwh

**Pull request merged**